components/tikv_util/src/memory.rs

```diff
@@ -8,6 +8,13 @@ use kvproto::{
 };
 use std::mem;
 
+// Transmute vec with types of same memory layout.
+pub unsafe fn vec_transmute<F, T>(from: Vec<F>) -> Vec<T> {
+    debug_assert!(mem::size_of::<F>() == mem::size_of::<T>());
```

Comment:
Better to also assert about the alignment.

And `static_assertions` may help check the error in compile time.

Reply:
can't, rust complains about it. Same goes for `static_assertions:: assert_eq_size`.
```
error[E0401]: can't use generic parameters from outer function
  --> components/tikv_util/src/memory.rs:14:56
   |
13 | pub unsafe fn vec_transmute<F, T>(from: Vec<F>) -> Vec<T> {
   |                             - type parameter from outer function
14 |     static_assertions::const_assert_eq!(mem::size_of::<F>(), mem::size_of::<T>());
   |                                                        ^ use of generic parameter from outer function
```
Seems no perfect solution for this, see https://stackoverflow.com/questions/30330519/compile-time-generic-type-size-check

Reply:
OK, I see.